src/main/java/io/strimzi/kafka/bridge/Application.java

```diff
@@ -88,70 +73,107 @@ public static void main(String[] args) {
-     * @param metricsReporter       MetricsReporter instance for scraping metrics from different registries
      * @return                      Future for the bridge startup
      */
-    private static Future<HttpBridge> deployHttpBridge(Vertx vertx, BridgeConfig bridgeConfig, MetricsReporter metricsReporter)  {
+    private static Future<HttpBridge> deployHttpBridge(BridgeConfig bridgeConfig) 
+            throws MalformedObjectNameException, IOException {
         Promise<HttpBridge> httpPromise = Promise.promise();
 
-        HttpBridge httpBridge = new HttpBridge(bridgeConfig, metricsReporter);
+        Vertx vertx = createVertxInstance(bridgeConfig);
+        MetricsCollector metricsCollector = getMetricsCollector(bridgeConfig);
+        HttpBridge httpBridge = new HttpBridge(bridgeConfig, metricsCollector);
+        
         vertx.deployVerticle(httpBridge)
-                .onComplete(done -> {
-                    if (done.succeeded()) {
-                        LOGGER.info("HTTP verticle instance deployed [{}]", done.result());
-                        httpPromise.complete(httpBridge);
-                    } else {
-                        LOGGER.error("Failed to deploy HTTP verticle instance", done.cause());
-                        httpPromise.fail(done.cause());
+            .onComplete(done -> {
+                if (done.succeeded()) {
+                    LOGGER.info("HTTP verticle instance deployed [{}]", done.result());
+                    if (metricsCollector != null) {
+                        LOGGER.info("Metrics of type '{}' enabled and exposed on /metrics endpoint", bridgeConfig.getMetricsType());
                     }
-                });
+                    httpPromise.complete(httpBridge);
+                } else {
+                    LOGGER.error("Failed to deploy HTTP verticle instance", done.cause());
+                    httpPromise.fail(done.cause());
+                }
+            });
 
         return httpPromise.future();
     }
 
+    private static Vertx createVertxInstance(BridgeConfig bridgeConfig) {
+        VertxOptions vertxOptions = new VertxOptions();
+        if (bridgeConfig.getMetricsType() != null) {
+            vertxOptions.setMetricsOptions(metricsOptions()); // enable Vertx metrics
+        }
+        Vertx vertx = Vertx.vertx(vertxOptions);
+        return vertx;
+    }
+
     /**
-     * Return a JmxCollectorRegistry instance with the YAML configuration filters
+     * Set up the Vert.x metrics options
      *
-     * @return JmxCollectorRegistry instance
-     * @throws MalformedObjectNameException
-     * @throws IOException
+     * @return instance of the MicrometerMetricsOptions on Vert.x
      */
-    private static JmxCollectorRegistry getJmxCollectorRegistry()
+    private static MicrometerMetricsOptions metricsOptions() {
+        Set<String> set = new HashSet<>();
+        set.add(MetricsDomain.NAMED_POOLS.name());
+        set.add(MetricsDomain.VERTICLES.name());
```

Comment:
What about using 

```java
Set<String> set = Set.of(MetricsDomain.NAMED_POOLS.name(), MetricsDomain.VERTICLES.name());
```

to replace the above three lines?

Reply:
Ok, I will also inline.